.github/workflows/ci.yml

```
@@ -157,3 +181,35 @@ jobs:
         with:
           flags: ${{ matrix.test-type }}
           env_vars: OS,PYTHON
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          # name has to be unique, to not overwrite uploads of other matrix runs. sha1 is optional and only to differentiate
+          # when locally dowloading and comparing results of different workflow runs.
+          name: test-results-${{ github.sha }}-macos-${{ matrix.python-version }}-${{ matrix.test-type }}-${{ matrix.test-group }}
+          path: |
+            .coverage
+            test-report.xml
+          retention-days: 1
+
+  analyze:
+    needs: [windows, linux, macos]
```

Let's be more descriptive here:

```suggestion
  analyze:
    needs: [windows, linux, macos]
    name: Analyze test results
```